src/components/pagination/index.tsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useState } from 'react'
+import {
+  StandaloneChevronLeftRegularIcon,
+  StandaloneChevronRightRegularIcon,
+} from '@deriv/quill-icons'
+import { usePaginationRange } from 'hooks/usePaginationRange'
+import PaginationButton from './pagination-button'
+import {
+  navigationIconFillVariants,
+  paginationVariants,
+} from './pagination.classnames'
+import { PaginationProps } from './pagination.types'
+
+/**
+ * Component that can divide large lists of data into smaller more manageable pages.
+ * @name Pagination
+ * @param {string} className- Styles to be applied to Pagination body
+ * @param {number} contentPerPage- Number of items to be displayed per page
+ * @param {Array} dataList- List of data to be displayed
+ * @param {React.Component} renderComponent- Component to be rendered
+ * @param {string} variant- Variant of Pagination.
+ *
+ * @example
+ * <Pagination
+ * className="pagination__container"
+ * contentPerPage={10}
+ * dataList={dataList}
+ * renderComponent={<DisplayContent />}
+ * variant="number"
+ * />
+ */
+const Pagination = <T,>({
+  className,
+  contentPerPage,
+  dataList = [],
+  renderComponent,
+  variant = 'number',
+}: PaginationProps<T>) => {
+  const [totalPageCount, setTotalPageCount] = useState(0)
+  const [currentPage, setCurrentPage] = useState(1)
+
+  const RenderComponent = renderComponent
+
+  const paginationRange = usePaginationRange({
+    totalPageCount,
+    currentPage,
+    variant,
+  })
+
+  useEffect(() => {
+    // Ensuring that 0 is not provided for contentPerPage
+    const dataToDisplay = contentPerPage > 0 ? contentPerPage : 1
+    setTotalPageCount(Math.ceil(dataList.length / dataToDisplay))
+  }, [dataList.length, contentPerPage])
```

Comment:
@likhith-deriv is it possible to replace this `useEffect` `useState` combination with `useMemo`? This may help avoid a few unnecessary rerenders.

Reply:
Let me implement and check